### PR TITLE
Provide config option to choose grpc or http exporters

### LINKED ISSUE #195: Provide config option to choose gRPC or HTTP exporters

### PR DESCRIPTION
**Pull request by LikeTheSalad**
Closes #195

### CODE REVIEW

**Review by bryce-b — DISMISSED** [review]

**Comment by ty-elastic** [comment]
looks great to me! thank you!!

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -104,6 +104,31 @@ class MyApp extends android.app.Application {
 }
 ----
 
+[float]
+[[server-export-protocol]]
+==== APM Server export protocol
+
+The signals will be exported using the gRPC protocol by default, you could change it to HTTP using the ExportProtocol configuration as shown below:
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:111]:
```suggestion
Signals are exported using the gRPC protocol by default. You can change the export protocol to HTTP by using the ExportProtocol configuration as shown below:
```

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -104,6 +104,31 @@ class MyApp extends android.app.Application {
...
+        super.onCreate();
+        ElasticApmConfiguration configuration = ElasticApmConfiguration.builder()
+                .setExportProtocol(ExportProtocol.HTTP) <1>
+                .build();
+        ElasticApmAgent.initialize(this, configuration);
+    }
+}
+----
+
+<1> Note that this configuration may not work if a <<custom-signal-configuration, custom signal configuration>> is set.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:128]:
```suggestion
----
<1> Note that this configuration may not work if a <<custom-signal-configuration, custom signal configuration>> is set.
```

File: docs/configuration.asciidoc [hunk 3]
```diff
@@ -104,6 +104,31 @@ class MyApp extends android.app.Application {
...
+
+NOTE: OTLP over HTTP support was added to the APM server on its version https://www.elastic.co/guide/en/apm/guide/current/release-notes-8.3.html#_added_8[8.3.0], whereas OTLP over gRPC is supported since version https://www.elastic.co/guide/en/apm/server/current/release-notes-7.12.html#_added_5[7.12.0].
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:130]: These are really old versions, so I don't think we need to link to the release notes here. I'd remove the links and just add the supported version.

```suggestion
NOTE: OTLP over HTTP is supported in APM server versions 8.3.0+. OTLP over gRPC is supported in APM Server versions 7.12.0+.
```

File: docs/configuration.asciidoc [hunk 4]
```diff
@@ -360,7 +386,43 @@ This table outlines the attributes and their corresponding permissions:
...
-==== Advanced configurable options
+=== Advanced configurable options
+
+[float]
+[[custom-signal-configuration]]
+==== Custom SignalConfiguration
+
+A SignalConfiguration object contains OpenTelemetry's processors for all the signals (spans, metrics and logs).
+The agent takes care of creating a default SignalConfiguration implementation based on the <<app-server-connectivity, Connectivity parameters>> passed during the agent's initialization, as well as the <<server-export-protocol, export protocol>> chosen for sending data to the APM server.
+However, if you need more control over OpenTelemetry's processors and exporters, you can override the default SignalConfiguration object and provide your own with custom processors and/or exporters as shown below:
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:397]:
```suggestion
A SignalConfiguration object contains OpenTelemetry's processors for all signals: spans, metrics, and logs).
The agent takes care of creating a default SignalConfiguration implementation based on the <<app-server-connectivity, Connectivity parameters>> passed during the agent's initialization, as well as the <<server-export-protocol, export protocol>> chosen for sending data to the APM Server.
However, if you need more control over OpenTelemetry's processors and exporters, you can override the default SignalConfiguration object and provide your own with custom processors and/or exporters, as shown below:
```

File: docs/configuration.asciidoc [hunk 5]
```diff
@@ -360,7 +386,43 @@ This table outlines the attributes and their corresponding permissions:
...
+
+        ElasticApmConfiguration.builder()
+                .setSignalConfiguration(mySignalConfiguration)
+                .build();
+        ElasticApmAgent.initialize(this, configuration);
+    }
+}
+----
+
+<1> You can either create your own implementation for the SignalConfiguration interface, or you could use the `SignalConfiguration.custom` function and pass your implementations for OpenTelemetry's processors and/or exporters.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:421]:
```suggestion
----
<1> You can either create your own implementation for the SignalConfiguration interface, or you can use the `SignalConfiguration.custom` function and pass your implementations for OpenTelemetry's processors and/or exporters.
```

**Review by bmorelli25 — DISMISSED** [review]
Just a few suggestions for the docs. Looks good!

**Review by AlexanderWert — APPROVED** [review]

**Pull request merged**